lib/cocoapods/installer.rb

```diff
@@ -563,6 +563,7 @@ def install_source_of_pod(pod_name)
     #
     def clean_pod_sources
       return unless installation_options.clean?
+      return if installed_specs.empty?
       pod_installers.each(&:clean!)
```

Comment:
will `pod_installers` not be empty when there are no installed specs?

Reply:
This is a very old PR so I had to check.

The `pod_installers` ivar contains all the installers, regardless of what gets (incrementally) generated.